metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,208 @@
+import sys
+
+from metaflow import R
+from metaflow.datastore.datastore import TransformableObject
+from metaflow.decorators import Decorator, StepDecorator
+from metaflow.exception import MetaflowException
+from metaflow.plugins.aws.awslambda.lambda_runner import LAMBDA_MAX_MEMORY_MB
+from metaflow.plugins.aws.batch.batch_decorator import (
+    DEFAULT_TIMEOUT_SECONDS,
+    BatchDecorator,
+    ResourcesDecorator,
+)
+from metaflow.plugins.aws.utils import sync_metadata_to_s3
+from metaflow.plugins.conda.conda_step_decorator import CondaStepDecorator
+from metaflow.plugins.timeout_decorator import get_run_time_limit_for_task
+
+if sys.version_info > (3, 0):
+    from typing import TYPE_CHECKING, Dict, List, Optional
+
+    if TYPE_CHECKING:
+        from metaflow.runtime import CLIArgs
+        from metaflow.graph import FlowGraph
+        from metaflow.package import MetaflowPackage
+        from metaflow.flowspec import FlowSpec
+        from metaflow.datastore import MetaflowDataStore
+        from metaflow.metaflow_environment import MetaflowEnvironment
+        from metaflow.event_logger import EventLogger
+
+from metaflow.metaflow_config import (
+    LAMBDA_IMAGE_URI,
+    LAMBDA_ROLE_ARN,
+)
+
+DEFAULT_TIMEOUT_SECONDS = 600
+
+
+class LambdaException(MetaflowException):
+    headline = "AWS Lambda error"
+
+
+class LambdaDecorator(StepDecorator):
+    """
+    The decorator is called @awslambda to avoid confusion, since "lambda" is a reserved
+    word in Python.
+    """
+
+    name = "awslambda"
+    defaults = {
+        "image": None,
+        "force_fetch_image": False,
```

Comment:
We can dispense with `force_fetch_image` - there is no particularly great way to handle this (I believe you want to handle the latest tag of the image here) and this issue is already present in AWS Batch/K8S etc. as well.

Reply:
Yes, this is mostly for the `:latest`. I'd have something similar for k8s and its `imagepullpolicy`, not sure how to handle AWS Batch tho.

Note that with Batch the problem is less acute, since you can hope that _eventually_ the new image will be used. With lambda, the image is copied by AWS when you create the function. So without this, there is just no way to trigger it to refetch the image, you'll be stuck with the old image forever. It was very annoying even when I was testing this on toy examples.